internal/services/kusto/kusto_database_data_source.go

```diff
@@ -0,0 +1,109 @@
+package kusto
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto/parse"
+	kustoValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func dataSourceKustoDatabase() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceKustoDatabaseRead,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.DatabaseID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: kustoValidate.DatabaseName,
+			},
+
+			"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+
+			"location": commonschema.LocationComputed(),
+
+			"cluster_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: kustoValidate.ClusterName,
+			},
+
+			"soft_delete_period": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"hot_cache_period": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"size": {
+				Type:     pluginsdk.TypeFloat,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceKustoDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Kusto.DatabasesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	clusterName := d.Get("cluster_name").(string)
+	name := d.Get("name").(string)
+	id := parse.NewDatabaseID(subscriptionId, resourceGroup, clusterName, name)
```

Comment:
This could be simplified a bit:
```suggestion
	id := parse.NewDatabaseID(subscriptionId, d.Get("resource_group_name").(string), d.Get("cluster_name").(string), d.Get("name").(string))
```